server/utils/utils.ts

```diff
@@ -166,3 +166,168 @@ export function getComponentString(macroName: string, params = {}): string {
 
   return nunjucks.renderString(macroString, {})
 }
+
+/**
+ * Adds missing question mark to a question.
+ *
+ * If a sentence is a question (based on the start of it) and it's missing
+ * the question marks it adds it. The input string is trimmed.
+ *
+ * Examples:
+ * - `'   DO BIRDS FLY  '` => `'DO BIRDS FLY?'`
+ * - `'When did this happen?'` => `'When did this happen?'` (unchanged, already ended with question mark)
+ * - `'Describe the incident'` => `'Describe the incident'` (unchanged, not a question)
+ * correctly (i.e. each part in a double-barreled is converted to proper case).
+ *
+ * @param label sentence to adjust
+ * @returns trimmed sented with a question mark if it's a question.
+ */
+export function addQuestionMarkToQuestion(label: string): string {
+  let result = label.trim()
+
+  const questionsPrefixes = [
+    'are',
+    'can',
+    'did',
+    'do',
+    'does',
+    'from what',
+    'from which',
+    'has',
+    'have',
+    'how',
+    'is',
+    'to which',
+    'was',
+    'were',
+    'what',
+    'when',
+    'where',
+    'which',
+    'who',
+    'whose',
+    'why',
+    'with what',
+  ]
+  const labelStart = result.toLowerCase()
+  if (questionsPrefixes.some(prefix => labelStart.startsWith(`${prefix} `))) {
+    if (!result.endsWith('?')) {
+      result += '?'
+    }
+  }
+
+  return result
+}
+
+/**
+ * Converts a string to sentence case
+ *
+ * First letter upper case, the rest in lower case. Preserve original
+ * case of acronyms and other special words.
+ *
+ * @param str input string
+ * @returns string in sentence case
+ */
+export function convertToSentenceCase(str: string): string {
+  // Special words (e.g. acronyms) - Preserve case if possible
+  const preserveList = [
+    'A1',
+    'ACCT',
+    'ADA',
+    'AFFF',
+    'AirPods',
+    'Amazon',
+    'Apple',
+    'BDH',
+    'C.I',
+    'C.N',
+    'C.S',
+    'CANDR',
+    'CCTV',
+    'CO2',
+    'CPS',
+    'CRC',
+    'DIMU',
+    'DJI',
+    'DMIU',
+    'DVD',
+    'DYI',
+    'EGS',
+    'F2052SH',
+    'F78A',
+    'FES',
+    'GP',
+    'H.C.C.',
+    'HCO',
+    'HHMD',
+    'HMPS',
```

Comment:
i think we need HMPPS here too even if it's not used yet

Reply:
Updated here: https://github.com/ministryofjustice/hmpps-incident-reporting/pull/287/commits/790cdaf0d52e8865428c4d05bd127067f162acdf